typing/env.ml

```diff
@@ -460,12 +460,20 @@ let read_pers_struct check modname filename =
   acknowledge_pers_struct check modname
     { Persistent_signature.filename; cmi }
 
+let can_load_cmis = ref true
+let without_cmis f x =
```

Comment:
I thought about using `Misc.try_finally`, but it seemed an overkill to create a closure for executing `can_load_cmis := true`. Yet there are many places where `try_finally` could be used.
I wasn't aware of `protect_refs`. This could be a good idea, as this is indeed more compact.

But what I want to know is just whether this pull request solves the problem at hand...